src/dialects/db2/data-types.ts

```diff
@@ -0,0 +1,380 @@
+      return `CHAR(${this._length}) FOR BIT DATA`;
+    }
+
+    if (this._length <= 4000) {
+      return `VARCHAR(${this._length}) FOR BIT DATA`;
+    }
+
+    return `BLOB(${this._length})`;
+  }
+
+  protected _hexify(hex: string) {
+    return `x'${hex}'`;
+  }
+
+  protected _stringify(
+    value: AcceptableTypeOf<BaseTypes.STRING>,
+    options: StringifyOptions,
+  ) {
+    if (this._binary) {
+      return this._hexify(value.toString('hex'));
+    }
+
+    return options.escape(value.toString());
+  }
+
+  protected _bindParam(
+    value: AcceptableTypeOf<BaseTypes.STRING>,
+    options: BindParamOptions,
+  ) {
+    return options.bindParam(this._binary ? Buffer.from(value) : value);
+  }
+}
+
+export class TEXT extends BaseTypes.TEXT {
+  // @ts-expect-error missing call to super() (useless here)
+  constructor() {
+    throw new Error('DB2 does not support the TEXT data-type');
+  }
+}
+
+export class BOOLEAN extends BaseTypes.BOOLEAN {
+  toSql() {
+    return 'BOOLEAN';
+  }
+
+  protected _sanitize(value: RawTypeOf<BaseTypes.BOOLEAN>) {
+    if (value !== null && value !== undefined) {
+      if (Buffer.isBuffer(value) && value.length === 1) {
+        // Bit fields are returned as buffers
+        value = value[0];
+      }
+
+      if (typeof value === 'string') {
+        // Only take action on valid boolean strings.
+        value = value === 'true' ? true : value === 'false' ? false : value;
+        value = value === '\u0001' ? true : value === '\u0000' ? false : value;
+      } else if (typeof value === 'number') {
+        // Only take action on valid boolean integers.
+        value = value === 1 ? true : value === 0 ? false : value;
+      }
+    }
+
+    return typeof value === 'boolean' ? value : null;
+  }
+
+  static readonly parse = this.prototype._sanitize;
+}
+
+export class UUID extends BaseTypes.UUID {
+  toSql() {
+    return 'CHAR(36) FOR BIT DATA';
+  }
+}
+
+export class NOW extends BaseTypes.NOW {
+  toSql() {
+    return 'CURRENT TIME';
+  }
+}
+
+export class DATE extends BaseTypes.DATE {
+  toSql() {
+    if (this._length != null) {
+      if (this._length < 0) {
+        this._length = 0;
+      }
+
+      if (this._length > 6) {
+        this._length = 6;
+      }
+    }
+
+    return `TIMESTAMP${this._length ? `(${this._length})` : ''}`;
+  }
+
+  protected _stringify(
+    date: AcceptableTypeOf<BaseTypes.DATE>,
+    options: StringifyOptions,
+  ) {
+    if (!moment.isMoment(date)) {
+      date = this._applyTimezone(date, options);
+    }
+
+    if (this._length != null && this._length > 0) {
+      let msec = '.';
+      for (let i = 0; i < this._length && i < 6; i++) {
+        msec += 'S';
+      }
+
+      return date.format(`YYYY-MM-DD HH:mm:ss${msec}`);
+    }
+
+    return date.format('YYYY-MM-DD HH:mm:ss');
+  }
+
+  static parse(value: RawTypeOf<BaseTypes.DATE>) {
+    if (typeof value !== 'string') {
+      // @ts-expect-error what type is value?
+      value = value.string();
```

Comment:
I have no idea what type this is.  In mysql this was the type the underlying lib returned and parse was called via mysql's `connection-manager` but it looked like db2 didn't use this (at least not in the same way mysql does) when I looked.